base58/base58.go

```diff
@@ -21,8 +19,8 @@ func Decode(b string) []byte {
 	j := big.NewInt(1)
 
 	for i := len(b) - 1; i >= 0; i-- {
-		tmp := strings.IndexAny(alphabet, string(b[i]))
-		if tmp == -1 {
+		tmp := b58[b[i]]
+		if tmp == 255 {
 			return []byte("")
 		}
 		idx := big.NewInt(int64(tmp))
```

Comment:
I'd actually like to remove most of the big math in this package, but that's a more invasive change.